omnigan/generator.py

```diff
@@ -87,6 +84,9 @@ def __init__(self, opts, latent_shape=None, verbose=None):
         if "w" in opts.tasks and not opts.gen.w.ignore:
             self.decoders["w"] = WaterDecoder(opts)
 
+        if "m" in opts.tasks and not opts.gen.m.ignore:
```

Comment:
I think you could remove "w" altogether from everywhere if `m` is going to do that job

Reply:
'w' was meant to be a water classifier, not a mask-generator right? Though I suppose one could make the argument that our discriminators are basically water classifiers

Reply:
W was supposed to tell the model where to flood, but in the translation scheme that also means actual water segmentation when flood->non-flood so if we drop that it's just the masker

Reply:
Ah. Ok then, farewell "WaterDecoder"